### PR TITLE
Incentive API Updates

### PR DESCRIPTION
**Pull request by anthonypetersen**
This PR addresses the following Issues:

* https://github.com/episphere/connect/issues/663
* https://github.com/episphere/connect/issues/664

Technical Changes:

* (Dashboard.js) - Removed option to authorize with site key
* (Firestore.js) - Removed function `validateSiteUser()`
* (Firestore.js) - Added `827220437` (Healthcare Provider) to return payload of `retrieveParticipantsEligibleForIncentives`
* (Firestore.js) - Updated function name and parameter name in `getChildrens()`
* (Shared.js) - Removed site key authorization checking from `APIAuthorization()`
* (Shared.js) - Added function `isDateTimeFormat()`
* (Incentive.js) - Refactored existing code
* (Incentive.js) - Added data checks on payload items
* (Incentive.js) - Added 206 return code option for data issues

### CODE REVIEW

**Review by JoeArmani — APPROVED** [review]
Checked items:
`dashboard.js`
-remove `siteKey` as a feature (remove siteKey check)
-return 401 if `!SSOObject`

`firestore.js`
-remove `validateSiteUser()` (coincides with removal of siteKey feature)
-add `return: Healthcare Provider` in `retrieveParticipantsEligibleForIncentives()`
-rename `getChildren` & update id variable casing

`incentive.js`
-destructure `data`
-handle errors with `token` and `round`
-handle & error handling -> `incentiveIssued` and `incentiveRefused`
-handle `caseNumber`
-update participantRecord(incentiveUpdates)
-refactor

`shared.js`
-remove `siteKey` auth logic & refs
-added DateTime format check (ISO 8601)

looks good

**Review by we-ai — APPROVED** [review]
Looks good.

File: utils/incentive.js [hunk 1]
```diff
@@ -1,86 +1,156 @@
...
+
+        const { token, round, incentiveIssued, incentiveIssuedAt, incentiveRefused, incentiveRefusedAt, incentiveChosen, caseNumber } = data[i];
+        
+        if(token === undefined) {
+            error = true;
+            responseArray.push({'Invalid Request': {'Token': 'UNDEFINED', 'Errors': 'Token not defined in data object.'}});
+            continue;
+        }
+
+        const { getParticipantData } = require('./firestore');
```
Comment (we-ai) [inline utils/incentive.js:34]: Better import functions outside the `for` loop, so that they're not repeatedly imported.

File: utils/incentive.js [hunk 2]
```diff
@@ -1,86 +1,156 @@
...
+            continue;
+        }
+
+        if(round !== 'baseline' && round !== 'followup1' && round !== 'followup2' && round !== 'followup3') {
+            error = true;
+            responseArray.push({'Invalid Request': {'Token': token, 'Errors': 'Invalid value for "Round".'}});
+            continue;
+        }
+
+        const { incentiveConcepts } = require('./shared');
```
Comment (we-ai) [inline utils/incentive.js:55]: Import before `for` loop.

**Pull request merged**